src/include/cpos.h

```diff
@@ -235,6 +245,57 @@ inline namespace _cpo {
 inline constexpr auto data = _data::_fn{};
 }  // namespace _cpo
 
+// ----------------------------------------------------------------------------
+// num_ids CPO
+// ----------------------------------------------------------------------------
+namespace _num_ids {
+void num_ids(auto&) = delete;
+void num_ids(const auto&) = delete;
+
+struct _fn {
+  template <class T>
+    requires(_member_num_ids<T>)
+  constexpr auto operator()(T&& t) const noexcept {
+    return t.num_ids();
+  }
+
+  template <class T>
+    requires(!_member_num_ids<T>)
+  constexpr auto operator()(T&& t) const noexcept {
+    return 0;
+  }
+};
+}  // namespace _num_ids
+inline namespace _cpo {
+inline constexpr auto num_ids = _num_ids::_fn{};
+}  // namespace _cpo
+
+// ----------------------------------------------------------------------------
+// ids CPO
+// @todo Figure out what is wrong with const
+// ----------------------------------------------------------------------------
+namespace _ids {
+void ids(auto&) = delete;
+void ids(const auto&) = delete;
+
+struct _fn {
+  template <class T>
+    requires(_member_ids<T>)
+  constexpr const auto& operator()(T&& t) const noexcept {
+    return t.ids();
+  }
+
+  template <class T>
+    requires(!_member_ids<T>)
+  constexpr const auto& operator()(T&& t) const noexcept {
+    return std::vector<typename std::remove_cvref_t<T>::value_type>{};
+  }
+};
+}  // namespace _ids
+inline namespace _cpo {
+inline constexpr auto ids = _ids::_fn{};
```

Comment:
Comment only -- we had offline discussion about whether or not this kind of cpo would conform with C++ convention.  Decided to leave it in for now.